pkg/scale/decode.go

```diff
@@ -506,7 +500,7 @@ func (ds *decodeState) decodeUint(dstv reflect.Value) (err error) {
 		buf := make([]byte, 3)
 		_, err = ds.Read(buf)
 		if err != nil {
-			return fmt.Errorf("reading bytes: %w", err)
+			return fmt.Errorf("reading byte: %w", err)
```

Comment:
change this back. We are reading more than one byte.
Such error would make sense on say `ReadByte`